clarifaipyspark/dataset.py

```diff
@@ -0,0 +1,371 @@
+    input_protos = []
+    input_obj = Inputs(user_id=self.user_id, app_id=self.app_id)
+
+    for row in dataframe.collect():
+      if labels:
+        labels_list = row["concepts"].split(',')
+        labels = labels_list if len(row['concepts']) > 0 else None
+      else:
+        labels = None
+
+      if 'metadata' in dataframe.columns:
+        if row['metadata'] is not None and len(row['metadata']) > 0:
+          metadata_str = row['metadata'].replace("'", '"')
+          try:
+            metadata_dict = json.loads(metadata_str)
+          except json.decoder.JSONDecodeError:
+            raise UserError("metadata column in CSV file should be a valid json")
+          metadata = Struct()
+          metadata.update(metadata_dict)
+        else:
+          metadata = None
+      else:
+        metadata = None
+
+      if 'geopoints' in dataframe.columns:
+        if row['geopoints'] is not None and len(row['geopoints']) > 0:
+          geo_points = row['geopoints'].split(',')
+          geo_points = [float(geo_point) for geo_point in geo_points]
+          geo_info = geo_points if len(geo_points) == 2 else UserError(
+              "geopoints column in CSV file should have longitude,latitude")
+        else:
+          geo_info = None
+      else:
+        geo_info = None
+
+      input_id = uuid.uuid4().hex
+      text = row["input"] if input_type == 'text' else None
+      image = row['input'] if input_type == 'image' else None
+      video = row['input'] if input_type == 'video' else None
+      audio = row['input'] if input_type == 'audio' else None
+
+      if df_type == 'raw':
+        input_protos.append(
+            input_obj.get_text_input(
+                input_id=input_id,
+                raw_text=text,
+                dataset_id=dataset_id,
+                labels=labels,
+                geo_info=geo_info))
+      elif df_type == 'url':
+        input_protos.append(
+            input_obj.get_input_from_url(
+                input_id=input_id,
+                image_url=image,
+                text_url=text,
+                audio_url=audio,
+                video_url=video,
+                dataset_id=dataset_id,
+                labels=labels,
+                geo_info=geo_info))
+      else:
+        input_protos.append(
+            input_obj.get_input_from_file(
+                input_id=input_id,
+                image_file=image,
+                text_file=text,
+                audio_file=audio,
+                video_file=video,
+                dataset_id=dataset_id,
+                labels=labels,
+                geo_info=geo_info))
+
+    return input_protos
+
+  def upload_dataset_from_dataframe(self,
+                            dataframe,
+                            input_type: str,
+                            df_type: str = None,
+                            labels: bool = True,
+                            chunk_size: int = 128) -> None:
+
+    if input_type not in ['image', 'text', 'video', 'audio']:
```

Comment:
```suggestion
    if input_type not in ('image', 'text', 'video', 'audio'):
```
A trivial one, better to use sets for faster lookups